src/include/daos/pool.h

```diff
@@ -53,6 +53,7 @@ struct dc_pool {
 	pthread_rwlock_t	dp_co_list_lock;
 	/* pool uuid */
 	uuid_t			dp_pool;
+	char		       *dp_label;
```

Comment:
I'm not sure. It may be possible but I think would require unwinding some code refactoring that was done. I think it is probably manageable here, but in a follow-on PR-5727 for container open bylabel, a struct dc_cont field "dc_label" is added too. There the refactoring is a little broader both in the dc_cont_open[_lbl]() functions and the completion callback cont_open_complete().

In this PR:
dc_pool_connect_internal() is used by both dc_pool_connect() and dc_pool_connect_lbl(). The internal function cannot call dc_task_get_args(task) to get an args structure, because there are 2 structure types now (daos_pool_connect_t and daos_pool_connect_lbl_t). So one way I found to refactor the code into this internal function was to store a label in struct dc_pool (gotten via dc_task_set/get_priv). We could decide to copy/paste the internal logic back into dc_pool_connect() and dc_pool_connect_lbl(). 

Other uses seem less of a challenge. e.g., all dc_pool_choose_svc_rank() calls for container and pool operations after a pool has been connected could simply specify a NULL label argument rather than pool->dp_label. i.e., the implementation would always choose a svc rank by providing UUID only.

dc_pool_choose_svc_rank() itself would need a small change too, to test whether the label argument is NULL, in a few places (i.e., avoid debug printing with %s if label is NULL).

Reply:
We could also add a label parameter to dc_pool_connect_internal, ..., or couldn't we?

Reply:
If you don't share the feeling about relabel support, then there's no need to make this change.

Reply:
I'm looking into adding label parameter to dc_pool_connect_internal() and also whether I can do something similar for the follow-on PR-5727 for container open bylabel (struct dc_cont similar issue).